reporter/otlp_reporter.go

```diff
@@ -215,37 +215,55 @@ func (r *OTLPReporter) ExecutableMetadata(args *ExecutableMetadataArgs) {
 	})
 }
 
+// FrameKnown return true if the metadata of the Frame specified by frameID is
+// cached in the reporter.
+func (r *OTLPReporter) FrameKnown(frameID libpf.FrameID) bool {
+	known := false
+	if frameMapLock, exists := r.frames.Get(frameID.FileID()); exists {
```

Comment:
The `r.frames` LRU should have a lifetime. Otherwise, `FrameMetadata()` is never going to be called again and symbols that have been removed in the backend will never appear again, except when the profiling agent is restarted or symbols drop out of the LRU.